storage/src/main/java/io/atomix/storage/journal/SegmentedJournalWriter.java

```diff
@@ -50,7 +50,6 @@ public long getNextIndex() {
   @Override
   public void reset(long index) {
     if (index > currentSegment.index()) {
-      currentWriter.close();
```

Comment:
@kuujo given reset is public wouldn't it be safer if we implemented something like this? https://github.com/atomix/atomix/compare/ensure_close?expand=1